### PR TITLE
feat(sca): Enable multiple image referencer framework results in the same scan

### PR DESCRIPTION
**Pull request by arielkru**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Before the PR, sca_image scan results would contain only one framework's result, instead of a list with all the image_cached_results from all image referencer frameworks. This change merges all the reports to be included in the final one.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/output/report.py [hunk 1]
```diff
@@ -577,6 +577,7 @@ def merge_reports(base_report: Report, report_to_merge: Report) -> None:
     base_report.failed_checks.extend(report_to_merge.failed_checks)
     base_report.skipped_checks.extend(report_to_merge.skipped_checks)
     base_report.parsing_errors.extend(report_to_merge.parsing_errors)
+    base_report.image_cached_results.extend(report_to_merge.image_cached_results)
```
Comment (gruebel) [inline checkov/common/output/report.py:580]: too easy 😄

**Review by gruebel — APPROVED** [review]
is it possible to add or extend a test for it?

**Comment by arielkru** [comment]
@gruebel good idea, added a test

**Review by gruebel — APPROVED** [review]
🍻

**Pull request merged**